### PR TITLE
Upload to azure from pip

### PR DESCRIPTION
**Pull request by philippmwirth**
# Upload to azure from pip

Adds the required headers for the PUT requests to the signed write-urls for AZURE.

### CODE REVIEW

**Review by MalteEbner — COMMENTED** [review]
I did not see tests for the new code, are you sure that it will be covered by the existing unittests? We will know for sure when the GitHub actions run again.

**Comment by philippmwirth** [comment]
> I did not see tests for the new code, are you sure that it will be covered by the existing unittests? We will know for sure when the GitHub actions run again.

It's almost impossible to test this exact case with the given setup in my opinion. Do you have an idea how to go about it?

**Comment by MalteEbner** [comment]
> It's almost impossible to test this exact case with the given setup in my opinion. Do you have an idea how to go about it?
You can have a `datasource_type == Azure` easily by overwriting `mocked_api_workflow_client._datasources_api.get_datasource_by_dataset_id()`.
You can just call `build_azure_signed_url_write_headers` with all parameters and ensure that values returned are the ones inserted.

Not testing `upload_file_with_signed_url` is ok for me.

File: lightly/api/api_workflow_client.py [hunk 1]
```diff
@@ -174,15 +174,19 @@ def upload_file_with_signed_url(self,
...
         """
+        if headers is not None:
+            response = requests.put(signed_write_url, data=file, headers=headers)
+        else:
+            response = requests.put(signed_write_url, data=file)
 
-        response = requests.put(signed_write_url, data=file)
-
-        if response.status_code != 200:
+        if response.status_code not in [200, 201]:
```
Comment (japrescott) [inline lightly/api/api_workflow_client.py:189]: nice one!

Comment (japrescott) [reply lightly/api/api_workflow_client.py:189]: possibly we could make this a response.status_code>=200 and response.status_code<300

**Review by japrescott — APPROVED** [review]

**Pull request merged**